apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.cs

```diff
@@ -80,26 +80,36 @@ public sealed partial class SpannerDbType
         /// </summary>
         public static SpannerDbType Numeric { get; } = new SpannerDbType(TypeCode.Numeric);
 
-        private static readonly Dictionary<TypeCode, SpannerDbType> s_simpleTypes
-            = new Dictionary<TypeCode, SpannerDbType>
+        /// <summary>
+        /// Representation of PostgreSQL numeric type. 
+        /// The PostgreSQL numeric type has max precision of 147,455 and a max scale of 16383.
+        /// </summary>
+        public static SpannerDbType PgNumeric { get; } = new SpannerDbType(TypeCode.Numeric, TypeAnnotationCode.PgNumeric);
+
+        private static readonly Dictionary<V1.Type, SpannerDbType> s_simpleTypes
+            = new Dictionary<V1.Type, SpannerDbType>
             {
-                {TypeCode.Unspecified, Unspecified },
-                {TypeCode.Bool, Bool },
-                {TypeCode.Int64, Int64 },
-                {TypeCode.Float64, Float64 },
-                {TypeCode.Timestamp, Timestamp },
-                {TypeCode.Date, Date },
-                {TypeCode.String, String },
-                {TypeCode.Bytes, Bytes },
-                {TypeCode.Json, Json },
-                {TypeCode.Numeric, Numeric }
+                { new V1.Type { Code = TypeCode.Unspecified } , Unspecified },
+                { new V1.Type { Code = TypeCode.Bool }, Bool },
+                { new V1.Type { Code = TypeCode.Int64 }, Int64 },
+                { new V1.Type { Code = TypeCode.Float64 }, Float64 },
+                { new V1.Type { Code = TypeCode.Timestamp }, Timestamp },
+                { new V1.Type { Code = TypeCode.Date }, Date },
+                { new V1.Type { Code = TypeCode.String }, String },
+                { new V1.Type { Code = TypeCode.Bytes }, Bytes },
+                { new V1.Type { Code = TypeCode.Json }, Json },
+                { new V1.Type { Code = TypeCode.Numeric }, Numeric },
+                { new V1.Type { Code = TypeCode.Numeric, TypeAnnotation = TypeAnnotationCode.PgNumeric }, PgNumeric }
             };
 
-        internal static SpannerDbType FromTypeCode(TypeCode code) 
-            => s_simpleTypes.TryGetValue(code, out SpannerDbType type) ? type : null;
+        internal static SpannerDbType FromType(V1.Type type) =>
+            s_simpleTypes.TryGetValue(type, out SpannerDbType dbType) ?
+            dbType : null;
 
         internal TypeCode TypeCode { get; }
 
+        internal TypeAnnotationCode TypeAnnotationCode { get; }
```

Comment:
At this point, it might even make sense to keep Type here. It's fine as it is though.